src/types.ts

```diff
@@ -1,20 +1,24 @@
-export type ArgoCDAppList = {
-  items: ArgoCDAppDetails[];
-}
+import * as t from 'io-ts';
 
-export type ArgoCDAppDetails = {
-  metadata: {
-    name: string;
-  };
-  status: {
-    sync: {
-      status: string;
-    };
-    health: {
-      status: string;
-    };
-    operationState: {
-      finishedAt: string;
-    };
-  };
-};
+export const argoCDAppDetails = t.type({

```

Comment:
io-ts library introduces a "codec" concept, which is a runtime equivalent of typescript types. It's a single source of truth, from which validators and typescript types are inferred.